scene/gui/popup_menu.cpp

```diff
@@ -402,8 +405,9 @@ void PopupMenu::_activate_submenu(int p_over, bool p_by_keyboard) {
 			}
 		}
 	}
-
 	submenu_pum->popup();
+	// Unpredictably, some submenus won't show even after popup without a process_frame.
+	get_tree()->connect("process_frame", callable_mp(this, &PopupMenu::_activate_submenu).bind(p_over, p_by_keyboard), CONNECT_ONE_SHOT);
```

Comment:
```suggestion
	callable_mp(this, &PopupMenu::_activate_submenu).bind(p_over, p_by_keyboard).call_deferred()
```

I wonder if deferring `submenu->popup()` would fix this, try `callable_mp(submenu_pum, &PopupMenu::popup).call_deferred();`

Reply:
I tried the call deferred per below and it doesn't prevent the "ghost" submenu issue like the `process_frame` statement:
```
callable_mp(submenu_pum, &PopupMenu::popup).bind(Rect2i(submenu_pos, submenu_size)).call_deferred();
```
What is the trick to yield for a `process_frame` or `RenderingServer.frame_post_draw`? I couldn't find a code example for this. I still suspect this is an issue with `Window` because `is_visible()` should never be true when it isn't.

Here's something I don't understand I'm curious about. In `_activate_submenu()` there is this statement:
```
	PopupMenu *submenu_pum = Object::cast_to<PopupMenu>(submenu_popup);
```
That recasts `PopupMenu` to another `Popupmenu`, which seems unneccessary. I tried skipping this step and using `submenu_popup` instead of `submenu_pum` and it works but the submenu won't resize correctly and has an extra margin at the bottom. Why is this step necessary and why does it fix that issue? Getting rid of it also doesn't make the ghosting issue go away, so that's not the source of the problem.

Reply:
Deferred calls happen at the end of the frame, while `process_frame` is sent at the beginning. This might be causing the difference.

Reply:
Okay, I removed the cobweb code where it was doing the duplicate cast to the same pointer, and did a few more tests looking at the calls to `_activate_submenu()`. Turns out that for those submenus that otherwise would not show, it took two additional calls to `_activate_submenu()` before they became visible. I think it's actually a reasonable solution at this point to use the one shot signal, until someone figures out what's happening in `Window` that prevents the popups from showing on the first call.